src/modules/accumulator.js

```diff
@@ -927,6 +927,49 @@ export default class AccumulatorModule extends WithParamsAndPublicKeys {
     }
   }
 
+  /**
+   * Update given witness by downloading necessary blocks and applying the updates if found. Both start and end are inclusive
+   * @param accumulatorId
+   * @param member
+   * @param witness - this will be updated to the latest witness
+   * @param startBlock - block number to start from
+   * @param endBlock - block number to end at. If not specified, it will pick the `lastUpdated` field of the accumulator.
+   * @returns {Promise<void>}
+   */
+  // eslint-disable-next-line sonarjs/cognitive-complexity
+  async updateVbAccumulatorWitnessFromUpdatesInBlocks(accumulatorId, member, witness, startBlock, endBlock = undefined) {
+    if (endBlock === undefined) {
+      const accum = await this.accumulatorModule.getAccumulator(accumulatorId, false);
+      // eslint-disable-next-line no-param-reassign
+      endBlock = accum.lastModified;
+    }
+    console.debug(`Will start updating witness from block ${startBlock} to ${endBlock}`);
+    let current = startBlock;
+    while (current <= endBlock) {
```

Comment:
it would be faster to step this in multiple increments rather than one at a time, eg:
- step from current to +5 blocks
- await Promise.all results from those 5 blocks, process results
- continue to step next 5 or less blocks until end

currently its waiting on promise results for one block at a time rather than "chunks" of block results